src/index.d.ts

```diff
@@ -58,6 +58,15 @@ export function inject<S, P, I, C>(
 // ):
 // <T extends IReactComponent<P & S>(target: T) => IReactComponent<P> & IWrappedComponent<T>
 
+/**
+ * disposeOnUnmount
+ */
+export function disposeOnUnmount(target: React.Component, propertyKey: string): void
```

Comment:
The official type for a field decorator is: `(target: Object, property: PropertyKey, descriptor?: PropertyDescriptor): void`

Reply:
I can change the others, but should the target be changed? as it is right now it prevents the decorator from being used on non react component classes

Reply:
Just took a look at lib.es5.d.ts and it shows
`declare type PropertyDecorator = (target: Object, propertyKey: string | symbol) => void;`
since the decorator should only work over react component classes and it doesn't work over symbol properties, shouldn't I leave it as is?